pkg/core/coretesting/resources.go

```diff
@@ -0,0 +1,45 @@
+package coretesting
+
+import (
+	"testing"
+
+	"github.com/klothoplatform/klotho/pkg/core"
+	"github.com/klothoplatform/klotho/pkg/graph"
+	"github.com/stretchr/testify/assert"
+)
+
+type (
+	StringDep = graph.Edge[string]
+
+	ResourcesExpectation struct {
+		Nodes []string
+		Deps  []StringDep
+
+		// AssertSubset assert the dag contains all the `.Nodes` and `.Deps`. If false,
+		// checks full equality.
+		AssertSubset bool
+	}
+)
+
+func (expect ResourcesExpectation) Assert(t *testing.T, dag *core.ResourceGraph) {
+	var res []string
+	for _, r := range dag.ListResources() {
+		res = append(res, r.Id())
+	}
+	if expect.AssertSubset {
```

Comment:
Aw, you didn't like the `matchFunc` idea? imo it makes it a bit more readable because it keeps branches out of the meat of the func, but definitely not a blocker.

Reply:
Mostly because I wanted the `ElementsMatch` ordering to match all the other asserts (ie, `assert(expected, actual)`) - even though it oddly uses `listA`, `listB` instead for some reason.

Reply:
Why is `listA, listB` incompatible with `expected, actual`?

Reply:
Mostly because the order is dictated by the `Subset(outer, inner)` where we pass in `Subset(dag, expected)` which is in the opposite order as `ElementsMatch(expected, dag)`.